### ISSUE TITLE
Unable to delete newly created workspace in new account

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#41357]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.68-0
**Reproducible in staging?:** Yes
**Reproducible in production?:** Yes
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @Puneet-here 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1714499352473189

## Action Performed:

1. Created a new account in staging.new.expensify.com
2. Create a new workspace
3. Select the created workspace 
4. Click three dot menu and delete the workspace

## Expected Result:
Workspace deleted
## Actual Result:
Unable to delete workspace and "Looks like you're on an annual subscription. Please downgrade your account from your account settings before trying again" error
## Workaround:
Unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [X] MacOS: Chrome / Safari
- [X] MacOS: Desktop

## Screenshots/Videos
![image (21)](https://github.com/Expensify/App/assets/38435837/73c421a3-cea8-42e4-a4c9-6ddbb372e834)

https://github.com/Expensify/App/assets/38435837/729b33b0-6485-4708-a5b6-100dde9f2e54

Add any screenshot/video evidence

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by trjExpensify** [comment]
I ran into this myself today.

1. I was able to delete the same workspace on OldDot, but not NewDot. 
2. I checked I was still in the 7-day free trial window

Logs are here (internal only, sorry). 

```
Throw ExpException - a8948deec2009297362f103209a7a026 ~~ message: '401 Can't delete last policy under a yearly subscription' exceptionMessage: 'Auth DeletePolicy returned an error' exceptionFile: '/git/releases/expensify.com/64247c4/lib/Auth.php' exceptionLine: '125' exceptionCode: '401'
```

I'm hitting this block when I don't think I should be: https://github.com/Expensify/Auth/blob/01554c45bd48e4cae90b02a23e96f943cb90698c/auth/lib/Policy.cpp#L2755-L2761

So I think somehow NewDot isn't recognising that the workspace I'm trying to delete is on a free trial and it's throwing this error.

**Comment by trjExpensify** [comment]
Sending external for more eyes.

**Comment by getusha** [comment]
Looks like a backend issue, i'll try to reproduce and confirm.

**Comment by dylanexpensify** [comment]
@getusha did we confirm?

**Comment by getusha** [comment]
I am unable to reproduce, @m-natarajan is there a prerequisite for the account?

https://github.com/Expensify/App/assets/75031127/d2a9efad-5f9b-4fb2-bbe4-71433a72a2c7

**Comment by dylanexpensify** [comment]
[bump](https://github.com/Expensify/App/issues/41357#issuecomment-2098365918) @m-natarajan

**Comment by isagoico** [comment]
@dylanexpensify This is still reproducible on our side. Applause Automation team reported seeing this issue ~2 hours ago

![image](https://github.com/Expensify/App/assets/44479856/1a4c9834-e48b-4a2a-ac03-992bddc5c5cf)

**Comment by dylanexpensify** [comment]
ah interesting, @MitchExpensify a collect workspace on a free trial be able to delete, right? Asking given your work with free trials

**Comment by trjExpensify** [comment]
Yep, and the bug is on NewDot only weirdly.

**Comment by MitchExpensify** [comment]
> a collect workspace on a free trial be able to delete, right? 

Correct

**Comment by getusha** [comment]
I think it's better to make this internal. @dylanexpensify

**Comment by dylanexpensify** [comment]
Moving internal.

**Comment by getusha** [comment]
Internal, not overdue

**Comment by dylanexpensify** [comment]
Still waiting

**Comment by dylanexpensify** [comment]
same

**Comment by carlosmiceli** [comment]
Commenting to follow and see if I can pick this one up soon.

**Comment by dylanexpensify** [comment]
@carlosmiceli let us know!

**Comment by dylanexpensify** [comment]
waiting to confirm if Carlos can pick up

**Comment by carlosmiceli** [comment]
Not yet, still have some stuff on my plate, but will continue to follow to see i i can pick up early next week.

**Comment by carlosmiceli** [comment]
Feel free to unassign me and add hot-pick if it makes sense!

**Comment by carlosmiceli** [comment]
Ok, should be able to get started tomorrow 💪

**Comment by dylanexpensify** [comment]
Nice! Let's get it @carlosmiceli!

**Comment by carlosmiceli** [comment]
@dylanexpensify @trjExpensify starting to work on this and get familiar with the code... Just so I understand, what's the expected behavior here? If the user is within the free trial he should be able to delete that last workspace, but not if trial is over? Or are there other subtleties to keep in mind? Because if it's simply the free trial check, I can add that [here](https://github.com/Expensify/Auth/blob/01554c45bd48e4cae90b02a23e96f943cb90698c/auth/lib/Policy.cpp#L2758) so it makes the exception during the free trial.

**Comment by trjExpensify** [comment]
@carlosmiceli the conditions for when you're able to delete a workspace should already exist in the code somewhere for a complete reference, as this is an action you can take on OldDot. It seems like the bug is on NewDot only when I ran into this: https://github.com/Expensify/App/issues/41357#issuecomment-2088612639

**Comment by carlosmiceli** [comment]
@trjExpensify Good point, but is it a BE issue then? If it's strictly FE, shouldn't this be External? I can still try to fix, but just to clarify.

**Comment by carlosmiceli** [comment]
First quick finding is that we are using different methods: `PolicyAPI::delete` in OD and `PolicyAPI::deleteWorkspace` in ND.

**Comment by trjExpensify** [comment]
>@trjExpensify Good point, but is it a BE issue then? If it's strictly FE, shouldn't this be External? I can still try to fix, but just to clarify.

I don't think so necessarily, it's just a bug that seems to only be happening on NewDot. Sounds like it's the diff you found here: 

PolicyAPI::delete in OD and PolicyAPI::deleteWorkspace in ND.

**Comment by dylanexpensify** [comment]
@carlosmiceli nice work getting PR up! Mind giving an ETA on merge?

**Comment by carlosmiceli** [comment]
@dylanexpensify My fix was actually not a good enough fix, there's more work to be done than I realized at first. I've been working on it but there's some things I still haven't figured out (asking for help [here](https://expensify.slack.com/archives/C03TQ48KC/p1719335935006529)), so not sure ETA yet 🙏

**Comment by carlosmiceli** [comment]
cc @dylanexpensify fyi, this is looking quite more complex, will keep you posted as I understand the best way to proceed: https://expensify.slack.com/archives/C03TQ48KC/p1719335935006529

**Comment by carlosmiceli** [comment]
Hey @dylanexpensify it seems like this issue will have to be put on hold because it depends on missing functionality that's supposedly being built (I'll ask and confirm). Conversation: https://expensify.slack.com/archives/C03TQ48KC/p1719335935006529

**Comment by dylanexpensify** [comment]
Sounds good, thanks Carlos!

**Comment by carlosmiceli** [comment]
I don't think there'll be an update for a while because it depends on this to be complete before we can properly know if it's a bug, a missing feature, or what exactly: https://github.com/Expensify/Expensify/issues/368528

**Comment by trjExpensify** [comment]
We merged the PR today to rollout the new subscription page btw.

**Comment by trjExpensify** [comment]
Subscription page hit prod yesterday, so taking the hold off. 👍

**Comment by dylanexpensify** [comment]
Nice! @carlosmiceli mind giving us an idea of when you'll be able to prioritize this?

**Comment by carlosmiceli** [comment]
Prioritizing the work for the Aug 1 deadline and some 1:1:1 refactoring these days, but I imagine  that this would come right after that! 🙏

**Comment by carlosmiceli** [comment]
Hopefully can start late this week or early next week.

**Comment by dylanexpensify** [comment]
We looking likely to kickstart this one this week, Carlos?

**Comment by dylanexpensify** [comment]
Nice nice, sounds good! Thanks for the update :)

**Comment by dylanexpensify** [comment]
@carlosmiceli how we looking here my man?

**Comment by carlosmiceli** [comment]
If things go really well, maybe I can start at the end of the week, but otherwise the second phase of the new homepage and some bugs we found in phase one are still priority... Sorry about that, obviously fine if someone with more bandwidth wants to take over in the meantime! If not this is my first thing after homepage is done. 🙏

**Comment by carlosmiceli** [comment]
Getting started with this!

**Comment by carlosmiceli** [comment]
Will need to pause to deal with some fixes for the homepage. Will pick up again as soon as I'm done with those if no one grabbed it.

**Comment by dylanexpensify** [comment]
Sounds good, we're hoping this will be done by EOW though for onshore goal. cc @garrettmknight

**Comment by carlosmiceli** [comment]
Yeah, I actually checked with him first because I wasn't sure what I should prioritize and he suggested what I said above. I'm gonna work on Saturday too anyway so hoping I still get to it (what else am I gonna do in the airport for a trillion hours?).

**Comment by carlosmiceli** [comment]
Still trying to get to it this week, as soon as the homepage bugs/improvements are done.

**Comment by carlosmiceli** [comment]
Getting started now!

**Comment by carlosmiceli** [comment]
This is looking like it's going to go back on HOLD until [this project](https://github.com/Expensify/Expensify/issues/399936) is complete (which would most likely take care of the issue in the process). Will confirm soon, here's the ongoing [conversation](https://expensify.slack.com/archives/C03TQ48KC/p1719335935006529) for context.

**Comment by garrettmknight** [comment]
Commented in [the thread](https://expensify.slack.com/archives/C03TQ48KC/p1725274486860609?thread_ts=1719335935.006529&cid=C03TQ48KC), but I agree that this issue should be held on the completion of the downgrades project... I also think it'd be great if @carlosmiceli led the detailed section of that project. 🙂

**Comment by trjExpensify** [comment]
Cool, sounds good. Let's stick a hold in the title and move it to be tracked with that downgrades project.

**Comment by carlosmiceli** [comment]
@trjExpensify Is that the right way to put the hold to the tracking issue?

**Comment by carlosmiceli** [comment]
@garrettmknight let me talk it out with Tim during my mentor session today and report back to see if it'd be a good idea, I ended up needing some help to craft the design doc for the landing page and that was a simpler project with a less urgent deadline, so I want to see what he thinks 🙏

**Comment by garrettmknight** [comment]
I'm not so sure that the downgrade project will be very complex. Give the [pre-design](https://expensify.slack.com/archives/C06ML6X0W9L/p1725292348720329) a look, but I think it's mostly FE work + updating the downgrade command to be 1:1:1.

**Comment by trjExpensify** [comment]
> @trjExpensify Is that the right way to put the hold to the tracking issue?

Yep, that does it! We can then downgrade this issue to `weekly` to check in on it until unblocked. Also, @garrettmknight @dylanexpensify prolly going to want to move this to #wave-control. I've put it in the project, but feel free to move it where it needs to be.

**Comment by carlosmiceli** [comment]
@garrettmknight Ok, I'm in! Read a bit yesterday, talked it out with @tgolen, and if we can talk it out a bit, I'm hoping I can rise to the challenge :)

Just need to make sure I wrap up the homepage project nicely. Will DM you to discuss further.

**Comment by dylanexpensify** [comment]
Pending Carlos to dig in!

**Comment by carlosmiceli** [comment]
To make sure we're on the same page, @dylanexpensify , this issue is now pending on this [project](https://github.com/Expensify/App/issues/41357#issuecomment-2324437683).

**Comment by carlosmiceli** [comment]
We've decided that this issue (reunification of Downgrading User to Free account) will _not_ be in the scope for the Workspace Downgrades project ([explanation](https://docs.google.com/document/d/1GDPmbvV47VVxucr-tHYjH80VylcJptP0RfGMVy3zanY/edit#bookmark=id.q4m27jkpzcrr)). It will be tackled in a separate follow-up project instead.

I'll move this to Monthly for now, and we can review each month if there's a project in the works to cover this, or if we should prioritize it and start one that month. cc @trjExpensify @dylanexpensify

**Comment by trjExpensify** [comment]
I'm confused what we're talking about, really. I think on OldDot we just look for whether you've paid us or not yet when you delete a workspace. If you haven't, we just let you delete the workspace without error. On NewDot, which I'm assuming is a different command, that hasn't been accommodated.

**Comment by garrettmknight** [comment]
From the OP, it looks like there was no way to change the subscription when this bug was created. Now that we have subscriptions in app, I don't think that's the case. In my video above, it shows that you can change subscriptions and delete the workspace... So I think this is 'solved' even if it's not super smooth or perfect.

<img width="850" alt="Screenshot 2024-09-23 at 12 36 20 PM" src="https://github.com/user-attachments/assets/41cdd213-238c-406f-b05f-f6a57a46c210">

**Comment by trjExpensify** [comment]
There's a workaround in NewDot now yeah, but it's not the expected result really. That being, when you click `Delete workspace` and you haven't subscribed to anything yet/paid, we just delete your workspace instead of throwing an error.

**Comment by dylanexpensify** [comment]
Yeah, I agree with a workaround existing. @trjExpensify @carlosmiceli @garrettmknight are we good to move forward with getting this done instead of closing it, yeah?

**Comment by trjExpensify** [comment]
I think Carlos is prioritising DowngradeToTeam first.

**Comment by dylanexpensify** [comment]
Reassigning as I head to parental leave next week! Thanks Al!

**Comment by alexpensify** [comment]
Weekly Update: Noted! I'll keep an eye out.

**Comment by alexpensify** [comment]
Weekly Update: On hold for internal 399936

**Comment by alexpensify** [comment]
Weekly Update: On hold

**Comment by alexpensify** [comment]
Weekly Update: 399936 is moving along but no update yet

**Comment by trjExpensify** [comment]
Moving this to #convert, per [here](https://expensify.slack.com/archives/C07NZ8B1VTQ/p1731019201368139).

**Comment by alexpensify** [comment]
I asked a question in the Slack 🧵 to confirm the `Hot Pick` status.

**Comment by carlosmiceli** [comment]
Just unassigned myself @alexpensify, if it's still happening when I'm done with a few issues I can take it back, but hopefully someone will fix it sooner :)

**Comment by danielrvidal** [comment]
I'll call it out for hot picks this week!

**Comment by alexpensify** [comment]
Weekly Update: Hot Pick

**Comment by mjasikowski** [comment]
I'll take this.

**Comment by mjasikowski** [comment]
Sorry for unassigning, the github assignee dropdown played some weird tricks on me.

**Comment by mjasikowski** [comment]
Allowing the policy to be deleted if it's during the free trial is pretty easy. But should there be any additional criteria? 

What should happen to the subscription afterwards (it's automatically set to yearly when on a free trial), should it be cancelled or stay as is? Are there any edge cases that we should think about here?

**Comment by carlosmiceli** [comment]
@mjasikowski hello sir, can you confirm that your fix would solve [this](https://github.com/Expensify/Expensify/issues/446101) issue?

**Comment by mjasikowski** [comment]
@carlosmiceli Can't confirm yet. I'll let you know when I figure out the complete workflow.

**Comment by mjasikowski** [comment]
@garrettmknight @carlosmiceli I made some changes to my draft PR. Let me know if the following solution works for you in principle:

1. When running the DeletePolicy command, `auth.policy.validateCanDelete` no longer will return an error when you are deleting the last paid policy while on the yearly plan, but only if the following conditions are met:
   - remaining policy has no items that should be billed (`shouldBillLastPolicy`)
   - the `allowDowngrade` function param is set to true (defaults to false, but will be true just for the DeletePolicy command)
   - account has no purchases (Account::canDowngrade(db, accountID))
2. If the above conditions are met, subscription type will be downgraded from yearly to monthly, and the policy will be deleted

There's no condition for free trials here, because free trial is triggered only if a billable action was made in a policy.
If you agree with the above process, I'll submit the PR to review.

**Comment by garrettmknight** [comment]
[Pulled into slack for some thoughts.](https://expensify.slack.com/archives/C07HPDRELLD/p1732721079045869?thread_ts=1732488812.057609&cid=C07HPDRELLD)

**Comment by alexpensify** [comment]
@mjasikowski—I'm back from Holiday and catching up. I reviewed the Slack conversation, and it looks like we are all set to continue here. Is there any update on when the PR will be ready? Thanks!

**Comment by mjasikowski** [comment]
@alexpensify the PR is already merged!

**Comment by alexpensify** [comment]
Whoops, sorry I missed it. Can you share the PR link here? I don't see one associated with this GH. Thanks!

**Comment by mjasikowski** [comment]
Sure thing, it's https://github.com/Expensify/Auth/pull/13280

**Comment by alexpensify** [comment]
@mjasikowski - since the PR was merged, can we close this one or are we waiting for another update? Thanks!

**Comment by mjasikowski** [comment]
@alexpensify let's close it!
